karbon/k8s_registration_structs.go

```diff
@@ -0,0 +1,102 @@
+package karbon
+
+type K8sCreateClusterRegistrationResponse struct {
+
+	// cluster name
+	// Required: true
+	ClusterName *string `json:"cluster_name"`
+
+	// cluster uuid
+	// Required: true
+	ClusterUUID *string `json:"cluster_uuid"`
+
+	// task uuid
+	// Required: true
+	TaskUUID *string `json:"task_uuid"`
+}
+
+type K8sClusterRegistrationDeleteResponse struct {
+
+	// Name of the k8s cluster being deleted.
+	ClusterName string `json:"cluster_name,omitempty"`
+
+	// UUID of the k8s cluster being deleted.
+	ClusterUUID string `json:"cluster_uuid,omitempty"`
+
+	// UUID of the task tracking the k8s cluster deletion.
+	// Required: true
+	TaskUUID *string `json:"task_uuid"`
+}
+
+type K8sClusterRegistration struct {
+
+	// Categories for this k8s cluster. This allows setting up multiple values from a single key.
+	CategoriesMapping map[string]string `json:"categories_mapping,omitempty"`
+
+	// identity
+	Identity *K8sIdentity `json:"identity,omitempty"`
+
+	// K8s cluster name.
+	// Required: true
+	// Max Length: 40
+	// Min Length: 1
+	Name *string `json:"name"`
+
+	// K8s cluster status.
+	Status string `json:"status,omitempty"`
+
+	// The universally unique identifier (UUID) of the k8s cluster.
+	// Pattern: ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
+	UUID string `json:"uuid,omitempty"`
+}
+
+type K8sClusterRegistrationList []*K8sClusterRegistration
+
+type K8sCreateClusterRegistrationRequest struct {
+
+	// Categories for this k8s cluster. This allows setting up multiple values from a single key.
+	CategoriesMapping map[string]string `json:"categories_mapping,omitempty"`
+
+	// identity
+	Identity *K8sIdentity `json:"identity,omitempty"`
+
+	// metadata
+	// Required: true
+	Metadata *Metadata `json:"metadata"`
+
+	// Unique name of the k8s cluster.
+	// Required: true
+	// Max Length: 40
+	// Min Length: 1
+	// Pattern: [a-z0-9]([-a-z0-9]*[a-z0-9])?
+	Name *string `json:"name"`
+
+	// The universally unique identifier (UUID) of the k8s cluster.
+	// Pattern: ^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$
```

Comment:
The patterns here won't make much sense on their own unless some sort of validation logic also accompanies them. We should consider adding the pattern validation logic client side.

Reply:
The validation logic already includes in acs-k8s 
```
func (m *K8sCreateClusterRegistrationRequest) validateUUID(formats strfmt.Registry) error {
                                                                                
        if swag.IsZero(m.UUID) { // not required                                
                return nil                                                      
        }                                                                       
                                                                                
        if err := validate.Pattern("uuid", "body", string(m.UUID), `^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$`); err != nil {
                return err                                                      
        }                                                                       
                                                                                
        return nil                                                              
}                   
```